src/test/scala/io/iohk/ethereum/consensus/validators/BlockHeaderValidatorSpec.scala

```diff
@@ -180,6 +177,39 @@ class BlockHeaderValidatorSpec
     res shouldBe Right(BlockHeaderValid)
   }
 
+  it should "mark as valid a post ecip1098 block opt-out with opt out defined" in new EphemBlockchainTestSetup {
+    val ecip1098BlockNumber = validBlockHeader.number / 2
+    val blockchainConfigWithECIP1098Enabled: BlockchainConfig = blockchainConfig.copy(ecip1098BlockNumber = ecip1098BlockNumber)
+    val blockHeaderValidator = new BlockValidatorWithPowMocked(blockchainConfigWithECIP1098Enabled)
+
+    val validHeader = validBlockHeader.copy(optOut = Some(true))
+
+    val validationResult = blockHeaderValidator.validate(validHeader, validParentBlockHeader)
+    validationResult shouldBe Right(BlockHeaderValid)
+  }
+
+  it should "mark as invalid a pre ecip1098 block opt-out with opt out defined" in new EphemBlockchainTestSetup {
+    val ecip1098BlockNumber = validBlockHeader.number * 2
+    val blockchainConfigWithECIP1098Disabled: BlockchainConfig = blockchainConfig.copy(ecip1098BlockNumber = ecip1098BlockNumber)
+    val blockHeaderValidator = new BlockValidatorWithPowMocked(blockchainConfigWithECIP1098Disabled)
+
+    val headerWithOptOutInvalidlyOn = validBlockHeader.copy(optOut = Some(true))
+
+    val validationResult = blockHeaderValidator.validate(headerWithOptOutInvalidlyOn, validParentBlockHeader)
+    validationResult shouldBe Left(HeaderOptOutError)
```

Comment:
(minor) maybe we could provide differenciation between this scenario and the following one, i mean in terms of `HeaderOptOutError` description. Still not sure if is worthy, wdyt?